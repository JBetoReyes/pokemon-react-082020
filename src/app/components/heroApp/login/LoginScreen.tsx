import React, { useContext } from 'react';
import { AppClickEvent } from '@typings/htmlEvents';
import { History } from 'history';
import AppContext, { IAppContext } from '../app.context';
import { logIn } from '../auth/auth.actions';

type Props = {
  history: History;
};

const LoginScreen = ({ history }: Props): JSX.Element => {
  const { user, setUser } = useContext(AppContext) as IAppContext;
  const btnWrapper = {
    width: '200px',
  };
  // eslint-disable-next-line @typescript-eslint/no-unused-vars
  const handleClick = (e: AppClickEvent) => {
    const lastPath = localStorage.getItem('lastPath') || '/dc';
    setUser(
      logIn({
        name: 'Beto',
        email: 'beto@emial.com',
        id: '1',
        logged: user.logged,
      })
    );
    history.push(lastPath);
  };
  return (
    <>
      <h1>Login Screen</h1>
      <hr />
      <div style={btnWrapper}>
        <button
          type="button"
          className="btn btn-outline-primary m-5 btn-block"
          onClick={handleClick}
        >
          Log in
        </button>
      </div>
    </>
  );
};

export default LoginScreen;
