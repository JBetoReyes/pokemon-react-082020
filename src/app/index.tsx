import React from 'react';
import ReactDOM from 'react-dom';
// import Home from './home/Home.component';
import App from './containers/App.component';
import '@styles/_main.scss';

ReactDOM.render(<App /> ,document.getElementById('app'));
