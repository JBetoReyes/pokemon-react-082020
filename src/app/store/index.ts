import { applyMiddleware, compose, createStore } from 'redux';
import thunk from 'redux-thunk';
import { IPokemon } from '../models/pokemonModel';

import rootReducer from './reducers';

const composeEnhancers =
  (typeof window !== 'undefined' &&
    (window as any).__REDUX_DEVTOOLS_EXTENSION_COMPOSE__) ||
  compose;
const pokemons: IPokemon[] = localStorage.getItem('pokemons')
  ? JSON.parse(localStorage.getItem('pokemons') as string)
  : [];
export default createStore(
  rootReducer,
  {
    exploreCarousel: {
      isloading: true,
      pokemons,
    },
  },
  composeEnhancers(applyMiddleware(thunk))
);
