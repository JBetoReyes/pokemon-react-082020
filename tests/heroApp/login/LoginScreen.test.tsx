import AppContext from '@components/heroApp/app.context';
import { IAppContext } from '@components/heroApp/app.model';
import LoginScreen from '@components/heroApp/login/LoginScreen';
import { mount } from 'enzyme';
import React from 'react';

describe('LoginScreen tests', () => {
  let appContext: IAppContext;
  let history;

  beforeEach(() => {
    appContext = {
      setUser: jest.fn(),
      user: {
        logged: false,
      },
    };
    history = {
      push: jest.fn(),
      replace: jest.fn(),
    };
  });
  afterAll(() => {
    jest.resetAllMocks();
  });

  test('Should match the snapshot', () => {
    const wrapper = mount(
      <AppContext.Provider value={appContext}>
        <LoginScreen history={history} />
      </AppContext.Provider>
    );
    expect(wrapper).toMatchSnapshot();
  });
  test('Should call the click handler', () => {
    const wrapper = mount(
      <AppContext.Provider value={appContext}>
        <LoginScreen history={history} />
      </AppContext.Provider>
    );
    wrapper.find('button').simulate('click');
    expect(appContext.setUser).toHaveBeenCalledWith({
      payload: {
        email: 'beto@emial.com',
        id: '1',
        logged: false,
        name: 'Beto',
      },
      type: '[auth] LOGIN',
    });
    expect(history.push).toHaveBeenCalledWith('/dc');
    localStorage.setItem('lastPath', '/marvel');
    wrapper.find('button').simulate('click');
    expect(history.push).toHaveBeenCalledWith('/marvel');
  });
});
