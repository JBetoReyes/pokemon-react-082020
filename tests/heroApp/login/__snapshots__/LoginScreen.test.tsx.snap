// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginScreen tests Should match the snapshot 1`] = `
<LoginScreen
  history={
    Object {
      "push": [MockFunction],
      "replace": [MockFunction],
    }
  }
>
  <h1>
    Login Screen
  </h1>
  <hr />
  <div
    style={
      Object {
        "width": "200px",
      }
    }
  >
    <button
      className="btn btn-outline-primary m-5 btn-block"
      onClick={[Function]}
      type="button"
    >
      Log in
    </button>
  </div>
</LoginScreen>
`;
